File: Back/src/main/kotlin/com/ssafy/tp/model/Product.kt
package com.ssafy.tp.model

import javax.persistence.*

@Entity(name = "product")
data class Product(

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name="id")
    var id: Long,

    @Column(name="product_name")
    var productName:String,

    @Column(name="product_img_path")
    var productImgPath:String,

    @Column(name="price")
    var price:Long,

    @Column(name="category")
    var category:String,

    @OneToMany
    @JoinColumn(name = "product_id")
    var reviews: List<Review>,


)

File: Flutter/lib/model/Product.dart
import 'dart:convert';

import 'package:main/model/Review.dart';

class Product {
  int? id;
  String? productName;
  int? price;
  String? category;
  String? productImgPath;
  List<dynamic>? reviews;

  Product({
    this.id,
    this.productName,
    this.price,
    this.category,
    this.productImgPath,
    this.reviews,
  });

  @override
  bool operator ==(other) {
    return (other is Product) && other.id == id && other.id == id;
  }

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
        id: json['id'],
        productName: json['productName'],
        price: json['price'],
        category: json['category'],
        productImgPath: json['productImgPath'],
        reviews: json['reviews']);
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'email': productName,
        'password': price,
        'phoneNumber': category,
      };
}
